### PR TITLE
Improve lua_api.txt a bunch

### LINKED ISSUE #10849: moveresult.touching_ground value is incorrect or misleading
##### Minetest version

```
Minetest 5.3.0 (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/nix/store/lyl5b6q7rvv7pp6szn0krddy7102fnx6-minetest-5.3.0/share/minetest"
STATIC_LOCALEDIR="/nix/store/lyl5b6q7rvv7pp6szn0krddy7102fnx6-minetest-5.3.0/share/locale"
```

I've also tested with commit `4fcd000e20a26120349184cb9d40342b7876e6b8` from January 21st.

##### First of all

Thanks for all the hard work. I'm loving writing mods for minetest.

##### Summary

The collision info table received during the `on_step` function has a wrong or misleading value for `touching_ground`. This value only seems to be `true` if the entity has just moved. For instance, you can spawn a physical entity in the air and set it's velocity to negative y. It will fall and `touching_ground` will be `false`, which is correct. Once it touches the ground, `touching_ground` becomes `true`, which is also correct, and the velocity seems to be reset to 0 because of the collision. The problem is that on the next `on_step`, `touching_ground` is now `false`, even though the entity is literally on the ground.

My guess is that collisions were not calculated because there is no movement, but somehow the engine didn't account for the fact that the entity is *already* on the ground.

This seems like a weird behavior to me, and I couldn't find a good way to determine that an entity is on the ground from Lua. Am I missing something here? If there's a good solution for this, I'm happy to write documentation about it.

### PR DESCRIPTION
**Pull request by sfan5**
inspired by my own woes

fixes #10849

### CODE REVIEW

**Comment by appgurueu** [comment]
~~We need more devs who actively use the API they develop.~~

**Review by NathanSalapat — CHANGES_REQUESTED** [review]
If you want to check the drawtype of a node, you could it do like this:

Probably meant,
If you want to check the drawtype of a node, you could do it like this:

File: doc/lua_api.txt [hunk 1]
```diff
@@ -7126,10 +7145,10 @@ object you are working with still exists.
     * in third person view (max. values `{x=-10/10,y=-10,15,z=-5/5}`)
 * `get_eye_offset()`: returns first and third person offsets.
 * `send_mapblock(blockpos)`:
-    * Sends a server-side loaded mapblock to the player.
-    * Returns `false` if failed.
+    * Sends an already loaded mapblock to the player.
+    * Returns `false` if nothing was sent (note that this can also mean that
```
Comment (rubenwardy) [inline doc/lua_api.txt:7149]: Wouldn't it be better to return true if the client already has the block

**Review by rubenwardy — APPROVED** [review]

File: doc/lua_api.txt (hunk 1 above)
Comment (sfan5) [reply doc/lua_api.txt:7149]: Yes, but I'm just describing the current state.

File: doc/lua_api.txt [hunk 2]
```diff
@@ -4553,10 +4549,10 @@ Callbacks:
       used for updating the entity state.
 * `on_deactivate(self)
     * Called when the object is about to get removed or unloaded.
-* `on_step(self, dtime)`
+* `on_step(self, dtime, moveresult)`
     * Called on every server tick, after movement and collision processing.
-      `dtime` is usually 0.1 seconds, as per the `dedicated_server_step` setting
-      in `minetest.conf`.
+    * `dtime`: elapsed time
```
Comment (SmallJoker) [inline doc/lua_api.txt:4554]:
```suggestion
    * `dtime`: elapsed time since last call
```

File: doc/lua_api.txt [hunk 3]
```diff
@@ -5695,13 +5714,13 @@ Item handling
...
+    * Example result for `"default:gold_ingot"` with two recipes:
 
           {
               [1]={method = "cooking", width = 3, output = "default:gold_ingot",
-              items = {1 = "default:gold_lump"}},
+              items = {[1] = "default:gold_lump"}},
               [2]={method = "normal", width = 1, output = "default:gold_ingot 9",
-              items = {1 = "default:goldblock"}}
+              items = {[1] = "default:goldblock"}}
           }
```
Comment (SmallJoker) [inline doc/lua_api.txt:5724]: Suggestion: width is 0 for shapeless recipes and explicit indices are not needed here. This version is longer but easier to read.

```
              {
                  method = "cooking", width = 0,
                  output = "default:gold_ingot",
                  items = {"default:gold_lump"}
              },
              {
                  method = "normal", width = 0,
                  output = "default:gold_ingot 9",
                  items = {"default:goldblock"}
              }
```

Comment (sfan5) [reply doc/lua_api.txt:5724]: Ah so the example was not even correct?

Comment (SmallJoker) [reply doc/lua_api.txt:5724]: Yes, multi-input cooking and hence width > 0 is currently not possible for the `"cooking"` method.
`"normal"` could have width > 0, but requires nested tables for the registration.

Comment (sfan5) [reply doc/lua_api.txt:5724]: Just checked, here's the real result in MTG:
```lua
{
	{
		output = "default:gold_ingot",
		items = {
			"keys:skeleton_key"
		},
		method = "cooking",
		width = 5,
		type = "cooking"
	},
	{
		output = "default:gold_ingot",
		items = {
			"keys:key"
		},
		method = "cooking",
		width = 5,
		type = "cooking"
	},
	{
		output = "default:gold_ingot",
		items = {
			"default:gold_lump"
		},
		method = "cooking",
		width = 3,
		type = "cooking"
	},
	{
		output = "default:gold_ingot 9",
		items = {
			"default:goldblock"
		},
		method = "normal",
		width = 1,
		type = "normal"
	}
}
```

Comment (SmallJoker) [reply doc/lua_api.txt:5724]: I see. `width` originates from `cooktime`, thanks to `CraftDefinitionCooking::getInput`.
Meanwhile the `cooktime` parameter is not documented at all.

Comment (sfan5) [reply doc/lua_api.txt:5724]: wat :thinking:

**Pull request merged**